### ISSUE TITLE
Unable to retrieve classes for MONDO

### ISSUE THREAD

**Issue by jvendetti** [ncbo/bioportal-project#125]
The REST API shows a class count of 120,941, but returns an empty collection (http://data.bioontology.org/ontologies/MONDO/classes). Attempting to navigate to the [Classes tab](http://bioportal.bioontology.org/ontologies/MONDO/?p=classes&conceptid=root) in the UI shows a "Problem retrieving classes" error.

Production parsing log file for submission 39 shows:

```
E, [2019-05-02T11:23:11.047941 #17925] ERROR -- : ["too many connection resets (due to end of file reached - EOFError) after 285252 requests on 47011693037600, last used 6924.860438528 seconds ago"]
E, [2019-05-02T11:23:11.048827 #17925] ERROR -- : [#<Net::HTTP::Persistent::Error: too many connection resets (due to end of file reached - EOFError) after 285252 requests on 47011693037600, last used 6924.860438528 seconds ago>]
E, [2019-05-02T11:23:11.050002 #17925] ERROR -- : ["NoMethodError: undefined method `id=' for nil:NilClass
/srv/ncbo/ncbo_cron/vendor/bundle/ruby/2.5.0/bundler/gems/ontologies_linked_data-548e7b1e4fb8/lib/ontologies_linked_data/models/ontology_submission.rb:1126:in `process_metrics'
	/srv/ncbo/ncbo_cron/vendor/bundle/ruby/2.5.0/bundler/gems/ontologies_linked_data-548e7b1e4fb8/lib/ontologies_linked_data/models/ontology_submission.rb:1058:in `process_submission'
	/srv/ncbo/ncbo_cron/lib/ncbo_cron/ontology_submission_parser.rb:177:in `process_submission'
	bin/ncbo_ontology_process:98:in `block in <main>'
	bin/ncbo_ontology_process:81:in `each'
	bin/ncbo_ontology_process:81:in `<main>'"]
```

**Comment by jvendetti** [comment]
_Notes on further investigation:_

The "too many connection resets" error occurs after the start of metrics calculation:

```
I, [2019-06-04T14:41:31.433349 #6962]  INFO -- : ["metrics_for_submission start"]
```

I ran the [ncbo_ontology_metrics](https://github.com/ncbo/ncbo_cron/blob/master/bin/ncbo_ontology_metrics) script in isolation on production, which finished quickly and without errors. However, the classes still aren't retrievable. I then attempted a total reprocessing of the ontology, but ended up with the same error as above. As soon as metrics calculation begins, the triplestore appears to be flooded with queries and hangs. 4store was unable to recover and had to be manually restarted.

I also ran a couple of experiments in our staging environment. Mondo is an ontology from the OBO Foundry and is available in a number of [different distribution formats](http://www.obofoundry.org/ontology/mondo.html). There's an OBO format equivalent, so I uploaded that and noted that there are no such issues:

http://stage.bioontology.org/ontologies/MONDOOBO?p=classes

Interestingly, I believe that the Ontology Lookup Service is also [serving the OBO](https://www.ebi.ac.uk/ols/ontologies/mondo) version, based on the appearance of the class tree.

Our staging environment was also able to handle the "minimal" OWL version of Mondo, which the foundry describes as being complete in terms of logical axioms, but has no textual definitions and subsets:

http://stage.bioontology.org/ontologies/MONDOMIN?p=classes

The REST endpoint returns the classes quickly since the results are paged. The UI will sometimes display an error message about taking too long to load the classes and only shows partial results. I opened the OWL version of the ontology in Protege and can see that there are a huge number of root classes with little no hierarchy - sort of like a bag of terms.

**Comment by jvendetti** [comment]
Uploaded the OBO version of this ontology to production so that end users have access to the ontology data until such time that we're able to diagnose/fix this issue with the OWL format. I had to do a "file upload" instead of entering a pull URL due to [this issue](https://github.com/ncbo/ncbo_cron/issues/23). This means that new versions of MONDO aren't getting pulled automatically.

**Comment by matentzn** [comment]
Maybe it has nothing to do with this, but just FYI:
https://github.com/EBISPOT/OLS/issues/312
https://github.com/monarch-initiative/mondo/issues/1451

**Comment by jvendetti** [comment]
There's been some udpates to the BioPortal software and the system is now able to handle / load the main OWL edition of Mondo (mondo.owl). Namely, @stdotjohn rewrote the code for calculating the ontology max depth metric in [version 1.4.0](https://github.com/ncbo/owlapi_wrapper/releases/tag/v1.4.0) of the owlapi_wrapper project. We've also transitioned from the defunct 4store triplestore to AllegroGraph.

I created a test entry today for the OWL version of MONDO in our staging environment and it loads without errors:

https://stage.bioontology.org/ontologies/MONDOOWL
